src/precice/impl/ParticipantImpl.cpp

```diff
@@ -514,6 +515,10 @@ void ParticipantImpl::finalize()
 
   // Finalize PETSc and Events first
   utils::Petsc::finalize();
+// This will lead to issues if we call finalize afterwards again
+#ifndef PRECICE_NO_GINKGO
+  device::Ginkgo::finalize();
```

Comment:
Consistent alternative here is to make this do nothing if precice isn't compiled with ginkgo.

Reply:
Depends a bit on how we treat the build pipeline: In the current pipeline, the kokkos compiler launcher takes a look at the corresponding source file, i.e., preciceCore only gets the link to `finalize`. If the library is then compiled without Ginkgo, the interface is not created and we end up with undefined references. We would then need to compile a dummy file.

Reply:
> We would then need to compile a dummy file.

True. This is essentially what we do for MPI and PETSc.

I am not saying one better than the other. Only upside in my eyes of the dummy is that calling code (especially ParticipantImpl) gets cleaner.

Reply:
Considering that we have `Ginkgo.cpp` now as a separate Object library, I would consider the current approach easier to realize.